src/data_info/services/trash.clj

```diff
@@ -66,40 +92,25 @@
                              (if-not (.startsWith path (paths/user-trash-path user))
                                {path (randomized-trash-path user path)}
                                {}))
-                           paths))]
-
-         (doseq [^String p paths]
-           (log/debug "path" p)
-           (log/debug "readable?" user (owns? cm user p))
-
-           ;;; Delete all of the tickets associated with the file.
-           (let [path-tickets (mapv :ticket-id (ticket-ids-for-path cm (:username cm) p))]
-             (doseq [path-ticket path-tickets]
-               (delete-ticket cm (:username cm) path-ticket)))
-
-           ;;; If the file isn't already in the user's trash, move it there
-           ;;; otherwise, do a hard delete.
-           (if (contains? trash-paths p)
-             (move-to-trash cm p (trash-paths p) user)
-             (delete cm p true))) ;;; Force a delete to bypass proxy user's trash.
-
+                           paths))
+             async-task-id (async-tasks/run-async-thread
+                             (rename/new-task "data-delete" user {:paths paths :trash-paths trash-paths})
```

Comment:
Eventually maybe `new-task` could be moved somewhere more common, like an `async-utils` namespace or something.

Reply:
Yeah, possibly. Right now, I keep it in `rename` because the idea is that this is the particular settings for a new task corresponding to a rename. Everything but "empty trash" in here works out to being a rename, basically, hence the reuse. It could be that we don't need to distinguish at all though, and then they could go into a utils namespace, or the async-tasks client namespace where I've put some other stuff (like `run-async-thread` above).